spec/views/curation_concerns/base/_form_progress.html.erb_spec.rb

```diff
@@ -64,4 +66,17 @@
       end
     end
   end
+
+  context "when the work has been saved before" do
+    before do
+      allow(work).to receive(:new_record?).and_return(false)
+      assign(:form, form)
+    end
+
+    let(:work) { stub_model(GenericWork, id: '456') }
```

Comment:
That's what I thought. But it does not matter what I do, it seems to be resolving `work` out of wack and picking up the wrong setting for `form.agreement_accepted`. It's probably because the setting is in the `form` (which depends on the `work`) and we are stubbing the `work` itself.